test/SugarChat.Database.MongoDb.IntegrationTest/MongoDbRepositoryTest.cs

```diff
@@ -9,52 +9,98 @@
 using System.Threading;
 using System;
 using System.Collections.Generic;
+using System.Linq;
 
 namespace SugarChat.Database.MongoDb.IntegrationTest
 {
-    public class MongoDbRepositoryTest : TestBase
+    public class MongoDbRepositoryTest : TestBase, IDisposable
     {
         readonly IRepository repository;
+        readonly Group _group;
         public MongoDbRepositoryTest()
         {
             MongoDbSettings settings = new MongoDbSettings();
             _configuration.GetSection("MongoDb")
                           .Bind(settings);
             repository = new MongoDbRepository(settings);
-            Should_Insert_One_Group().Wait();
+            _group = new Group()
+            {
+                AvatarUrl = "https://Avatar.jpg",
+                CreatedBy = Guid.NewGuid().ToString(),
+                CreatedDate = DateTime.Now,
+                CustomProperties = new Dictionary<string, string>(),
+                Description = "A Test Group!",
+                Id = Guid.NewGuid().ToString(),
+                LastModifyDate = DateTime.Now,
+                LastModifyBy = Guid.NewGuid().ToString(),
+                Name = "TestGroup"
+            };
+            repository.AddAsync(_group).Wait();
         }
 
         [Fact]
-        public void Should_Repository_Not_Null()
+        public async Task Should_Insert_One_Group_And_Remove_It()
         {
-            repository.ShouldNotBeNull();
+            var newGroup = new Group()
+            {
+                AvatarUrl = "https://Avatar.jpg",
+                CreatedBy = Guid.NewGuid().ToString(),
+                CreatedDate = DateTime.Now,
+                CustomProperties = new Dictionary<string, string>(),
+                Description = "A Test Group!",
+                Id = Guid.NewGuid().ToString(),
+                LastModifyDate = DateTime.Now,
+                LastModifyBy = Guid.NewGuid().ToString(),
+                Name = "NewTestGroup"
+            };
+            await repository.AddAsync(newGroup, default(CancellationToken));
+            var firstGroup = await repository.FirstOrDefaultAsync<Group>(e => e.Id == newGroup.Id);
+            firstGroup.ShouldNotBeNull();
+            await repository.RemoveAsync(newGroup);
```

Comment:
If previous asserts failed, the following remove operation will not be executed as expected, I guess a try...finally block will work better here.